pkg/performanceprofile/cmd/render/render.go

```diff
@@ -188,45 +190,96 @@ func render(inputDir, outputDir string) error {
 			return err
 		}
 
-		uid := pp.UID
-		if uid == types.UID("") {
-			uid = uuid.NewUUID()
-		}
-
-		or := []v1.OwnerReference{
-			{
-				Kind:       pp.Kind,
-				Name:       pp.Name,
-				APIVersion: pp.APIVersion,
-				UID:        uid,
-			},
-		}
-
-		for _, componentObj := range components.ToObjects() {
-			componentObj.SetOwnerReferences(or)
+		if addOwnerRef {
+			err = addOwnerReference(components, pp)
+			if err != nil {
+				return err
+			}
 		}
-
-		for kind, manifest := range components.ToManifestTable() {
-			b, err := yaml.Marshal(manifest)
+		if len(addAnnRef) > 0 {
+			addWeakOwnerReferenceAnnotation(components, pp, addAnnRef)
+			err = writeObject(outputDir, fmt.Sprintf("%s_%s.yaml", pp.Name, "annotated"), pp)
 			if err != nil {
 				return err
 			}
+		}
 
-			fileName := fmt.Sprintf("%s_%s.yaml", pp.Name, strings.ToLower(kind))
-			fullFilePath := filepath.Join(outputDir, fileName)
-			klog.Info("Writing file: ", fullFilePath)
-
-			err = os.WriteFile(fullFilePath, b, 0644)
+		for kind, manifest := range components.ToManifestTable() {
+			err = writeObject(outputDir, fmt.Sprintf("%s_%s.yaml", pp.Name, strings.ToLower(kind)), manifest)
 			if err != nil {
 				return err
 			}
-			klog.Info(fileName)
 		}
 	}
 
 	return nil
 }
 
+func writeObject(outputDir, fileName string, manifest interface{}) error {
+	fullFilePath := filepath.Join(outputDir, fileName)
+	klog.Infof("Writing file: %s -> %s", fileName, fullFilePath)
+
+	b, err := yaml.Marshal(manifest)
+	if err != nil {
+		return err
+	}
+
+	return os.WriteFile(fullFilePath, b, 0644)
+}
+
+func addOwnerReference(components *manifestset.ManifestResultSet, pp *performancev2.PerformanceProfile) error {
+	if pp == nil || pp.UID == types.UID("") {
+		return fmt.Errorf("Missing UID from performance profile")
+	}
+
+	or := []v1.OwnerReference{
+		{
+			Kind:       pp.Kind,
+			Name:       pp.Name,
+			APIVersion: pp.APIVersion,
+			UID:        pp.UID,
+		},
+	}
+
+	for _, componentObj := range components.ToObjects() {
+		componentObj.SetOwnerReferences(or)
+	}
+
+	return nil
+}
+
+const (
+	weakOwnerReferenceAnnotation = "performance.openshift.io/weak-owner-reference"
+)
+
+func addWeakOwnerReferenceAnnotation(components *manifestset.ManifestResultSet, pp *performancev2.PerformanceProfile, addAnnRef string) {
+	ref := makeWeakOwnerRef(addAnnRef, pp.UID)
+
+	if pp.Annotations == nil {
+		pp.Annotations = make(map[string]string)
+	}
+	pp.Annotations[weakOwnerReferenceAnnotation] = ref
```

Comment:
I wonder if label is better here. As it will allow searching later when we re-link the objects together.

Reply:
fair point, I'll have a look